Add ansible_uptime_seconds support for AIX

schroeert opened the pull request:
##### SUMMARY

AIX was missing the ansible_uptime_seconds fact, which is available for Linux.
I implemented get_uptime_facts for AIX.

##### ISSUE TYPE

- Feature Pull Request

##### ADDITIONAL INFORMATION

The ansible.builtin.setup module collects various facts, such as ansible_uptime_seconds. This fact is implemented for most operating systems, with the exception of AIX.

schroeert committed:
Update aix.py

Added ansible_uptime_seconds fact for AIX

neikei reviewed (approved):
/lgtm - Many thanks for pushing content for AIX 🚀

schroeert committed:
Create 84321-added-ansible_uptime_seconds_aix

schroeert committed:
Rename 84321-added-ansible_uptime_seconds_aix to 84321-added-ansible_uptime_seconds_aix.yml

Fixed linting issue

ERROR: changelogs/fragments/84321-added-ansible_uptime_seconds_aix:0:0: extension must be one of: .yml, .yaml (75%)

schroeert committed:
Update aix.py

Fixed linting issues

01:45 ERROR: Found 7 pep8 issue(s) which need to be resolved:
01:45 ERROR: lib/ansible/module_utils/facts/hardware/aix.py:130:78: W291: trailing whitespace (100%)
01:45 ERROR: lib/ansible/module_utils/facts/hardware/aix.py:131:76: W291: trailing whitespace (100%)
01:45 ERROR: lib/ansible/module_utils/facts/hardware/aix.py:137:40: E231: missing whitespace after ',' (100%)
01:45 ERROR: lib/ansible/module_utils/facts/hardware/aix.py:138:17: E225: missing whitespace around operator (100%)
01:45 ERROR: lib/ansible/module_utils/facts/hardware/aix.py:139:18: E225: missing whitespace around operator (100%)
01:45 ERROR: lib/ansible/module_utils/facts/hardware/aix.py:140:20: E225: missing whitespace around operator (100%)
01:45 ERROR: lib/ansible/module_utils/facts/hardware/aix.py:141:20: E225: missing whitespace around operator (100%)

Akasurde commented on lib/ansible/module_utils/facts/hardware/aix.py:
```diff
@@ -123,6 +125,26 @@ def get_memory_facts(self):
 
         return memory_facts
 
+    def get_uptime_facts(self):
+        uptime_facts = {}
+        # On AIX, there are no options to get the uptime directly in seconds.
+        # Your options are to parse the output of "who", "uptime", or "ps".
+        # Only "ps" always provides a field with seconds.
+        rc, out, err = self.module.run_command("/usr/bin/ps -p 1 -o etime=")
+        # Parse out
+        if out:
+            lines = out.splitlines()
+            data = lines[0].replace(':', '-').split('-')
+            days = int(data[0])
+            hours = int(data[1])
+            minutes = int(data[2])
+            seconds = int(data[3])
+            # Calculate uptime in seconds
+            uptime_seconds = (days * 86400) + (hours * 3600) + (minutes * 60) + seconds
+            uptime_facts['uptime_seconds'] = int(uptime_seconds)
+
+        return uptime_facts
+
```
```suggestion
    def get_uptime_facts(self):
        uptime_facts = {}
        # On AIX, there are no options to get the uptime directly in seconds.
        # Your options are to parse the output of "who", "uptime", or "ps".
        # Only "ps" always provides a field with seconds.
        ps_bin = self.module.get_bin_path("ps")
        if ps_bin is None:
            return uptime_facts
        
        ps_cmd = [ps_bin, "-p", "1", "-o", "etime="]
        
        rc, out, err = self.module.run_command(ps_cmd)
        if rc != 0:
            return uptime_facts

        # Parse out
        if out:
            lines = out.splitlines()
            data = lines[0].replace(':', '-').split('-')
            try:
                days = int(data[0])
                hours = int(data[1])
                minutes = int(data[2])
                seconds = int(data[3])
            except (IndexError, ValueError):
                return uptime_facts
            # Calculate uptime in seconds
            uptime_seconds = (days * 86400) + (hours * 3600) + (minutes * 60) + seconds
            uptime_facts['uptime_seconds'] = int(uptime_seconds)

        return uptime_facts

```

Akasurde reviewed (changes requested):
Add an unit test for this change. 
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

schroeert committed:
Update lib/ansible/module_utils/facts/hardware/aix.py

Tested change on AIX. Result as expected.

Co-authored-by: Abhijeet Kasurde <akasurde@redhat.com>

schroeert committed:
Update aix.py

Removed trailing-whitespace.

schroeert committed:
Update 84321-added-ansible_uptime_seconds_aix.yml

Changed changelog fragment as requested.

schroeert commented:
Added the changelog_fragment and tested the provided changes on AIX. I'm not sure why the checks failed. There seems to be some issue with the Pipeline.
```
00:36 WARNING: 400: invalid: Azure Pipelines job ansible/ansible:128783.1:Remote.test_macos_14_3_1.__default public key 'ansible-core-ci/public-key.pem' not found.

webknjaz commented:
Restarted the failing jobs.
